python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1900,6 +1905,108 @@ def get_event_records(
         """
         return self._event_storage.get_event_records(event_records_filter, limit, ascending)
 
+    @public
+    @traced
+    def get_materialization_records(
+        self,
+        records_filter: Optional[Union[AssetKey, "AssetRecordsFilter"]] = None,
+        limit: int = 10000,
```

Comment:
I think we should enforce an upper limit (thats backed by an env var so can be configured per deployment) thats this 10,000

Did you consider having `limit` not be defaulted and forcing every call site to specify something? 

If we do want to have a default limit, i think it should be lower then the upper bound 10k, maybe like 1k? 100? But really i think it would be preferable to force call sites to specify

Reply:
Yeah, I think enforcing the caller to specify `limit` makes sense.